mitodl/open-discussions
# Add support for multiple Offered By

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #2163 

#### What's this PR do?
- Adds a many-to-many relation from `LearningResource` subclasses (in practice, only `Course`, `LearningResourceRun`, and `Program`) to a new `LearningResourceOfferor`, which represents which platform(s) are offering.
- Updates existing ingestion mechanisms to write the new records, not that for the `course_catalog.etl` packages, this is additive-only, so that MITx and MicroMasters can both add themselves as an offeror without stomping on the other.
- Updated indexing serializers to handle the new data structure
- Frontend was updated to handle the new data structure
  - If a resource is available on both MicroMasters and MITx, MicroMasters is given precedence
- Adds a missing backpopulate command for MicroMasters

#### How should this be manually tested?
- Run migrations and reindex (keep in mind that because the was the existing data and migrations work, resources will only have a single item in `offered_by`)
- Verify the search page still works
- Create a course that exists both on MITx and MM, run backpopulate on each and verify that the course has two offerors and that "MicroMasters" is displayed both in the search card and the drawer on the "Take Course on..." button
